lib/sisimai/bite/email/activehunter.rb

```diff
@@ -0,0 +1,124 @@
+module Sisimai::Bite::Email
+  module Activehunter
+    # Sisimai::Bite::Email::Activehunter parses a bounce email which created
+    # by TransWARE Active!hunter.
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/Activehunter.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :from    => %r/\A"MAILER-DAEMON"/,
+        :subject => %r/FAILURE NOTICE :/,
+      }
+      Re1 = {
+        :begin   => %r/\A  ----- The following addresses had permanent fatal errors -----\z/,
+        :error   => %r/\A  ----- Transcript of session follows -----\z/,
+        :rfc822  => %r|\AContent-type: message/rfc822\z|,
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'TransWARE Active!hunter'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['X-AHMAILID']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from TransWARE Active!hunter
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['x-ahmailid']
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        v = nil
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [50.13/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->